bin/security_scan

```diff
@@ -0,0 +1,64 @@
+#!/usr/bin/env bash
+
+# This script is designed to be ran within the Secretless-Broker repository,
+# to use elsewhere, you will need to change this.
+working_directory=${PWD}/..
+
+# Get an array of directories containing modified files
+modified_directories=$(git diff --name-only | xargs -L1 dirname | uniq)
+
+# Exclude test files
+test_dir=${working_directory}/test
+
+print_usage() {
+    echo "Security Scanner"
+    echo " "
+    echo "Description:"
+    echo "Runs gosec on directories which git detects and marks in the diff."
+    echo "If the branch is detected as 'master', it will scan all"
+    echo "directories regardless of what has been modified locally."
+    echo " "
+    echo "Format:"
+    echo "security_scan [arguments]"
```

Comment:
Since we wanted to run it within a docker container, I thought it would be easiest to have two separate files. I'm not very familiar with bash, but I assume that's the easiest way to run a large script like this inside a docker run command?

Reply:
ok, so there are two scripts because one is meant to be run in a docker container (which docker container?) and the other one actually runs the the container?

I feel like there might be some patterns for how to name and organize these that would make their intended use more clear, but I'm not close enough to the code lately to remember exactly what to point you to. @sgnn7 @jonahx do you have any examples of how we've done something like this in other cases? in particular, I'm concerned that if I look at the `bin/` dir, it's not going to be obvious which script I should run to do a `gosec` scan if we have two with similar names. I don't think this is the first time we've had to think about something like this.

Reply:
Currently I have them named as such:
- check_security: performs the gosec scan with whatever parameters you need
- run_check_security: creates a docker container with secretless mounted as a volume, and runs check_security on the volume